frontend/src/components/Question.vue

```diff
@@ -2,29 +2,24 @@
 import { computed, defineAsyncComponent, ref } from 'vue';
 import type { ComputedRef, Ref } from 'vue';
 import type { Quizoot } from '@interfaces/quizoot';
-import NavigationButton from '@/components/NavigationButton.vue';
+import type { Question } from '@interfaces/quizoot.indexed';
+import { useFetch } from '@/lib/hooks';
 import { pascalToSnake } from '@/lib/string-utils';
-
-interface QuestionsFlow {
-    questionNumber: number;
-    isFirstQuestion: boolean;
-    isLastQuestion: boolean;
-}
+import FetchError from '@/components/FetchError.vue';
+import Loader from '@/components/Loader.vue';
 
 interface QuestionWrapperProps {
-    question: Quizoot.Question;
-    questionsFlow: QuestionsFlow;
+    questionId: Quizoot.Question['id'];
+    questionNumber: number;
     questionsCount: number;
-    goToPreviousQuestion: () => void;
-    goToNextQuestion: () => void;
 }
 
-type Components = Promise<any>;
-type ComponentsMap = Record<Quizoot.QuestionKind, Components>;
+type Component = Promise<any>;
```

Comment:
we could type this better right ?

Reply:
I saw two options:
```ts
// 1.
type Component = Promise<object>;

// 2
type Component = typeof defineAsyncComponent;
```

I think the second one is the best.